### PR TITLE
bug: ENV variable SE_VNC_PASSWORD contains sensitive data

### LINKED ISSUE #2060: [🐛 Bug]:  The environment variable SE_VNC_PASSWORD contains sensitive data
### What happened?

In https://github.com/SeleniumHQ/docker-selenium/pull/2056 an environment variable `SE_VNC_PASSWORD` was added which is flagged by security scanners. The first release with this commit is https://github.com/SeleniumHQ/docker-selenium/releases/tag/4.16.1-20231212

If I create the image locally as is with `ENV SE_VNC_PASSWORD=secret` the secret is flagged. If I create with `ENV SE_VNC_PASSWORD=` then it is not flagged. Is it possible to set the variable without a default value?

### Command used to start Selenium Grid with Docker (or Kubernetes)

```shell
This is an environment variable issue
```

### Relevant log output

```shell
This is an environment variable issue
```

### Operating System

all

### Docker Selenium version (tag or chart version)

4.16.1-20231212

### PR DESCRIPTION
**Pull request by VietND96**
**Thanks for contributing to the Docker-Selenium project!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://selenium.dev/documentation/en/contributing/) guidelines, applied for this repository.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description
bug: ENV variable SE_VNC_PASSWORD contains sensitive data

### Motivation and Context
- Fix #2060 
- Address high vulnerability raised by Trivy workflow (added via https://github.com/SeleniumHQ/docker-selenium/commit/93da2b58011bfb508c59d212ce428c6716896752)
- Update dependencies version same as the upstream project as per comment https://github.com/SeleniumHQ/docker-selenium/pull/2059#discussion_r1423808839

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://selenium.dev/documentation/en/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by diemol — COMMENTED** [review]
What is the actual change I should review? I see many unrelated things to the VNC password.

**Comment by luisfcorreia** [comment]
From what I've read, the tool considers that an ENV setting containing a name with "PASSWORD" in it is a security concern.

For me, it's NOT

File: Base/Dockerfile [hunk 1]
```diff
@@ -4,8 +4,9 @@ LABEL authors="Selenium <selenium-developers@googlegroups.com>"
 # Arguments to define the version of dependencies to download
 ARG VERSION
 ARG RELEASE=selenium-${VERSION}
-ARG OPENTELEMETRY_VERSION=1.31.0
-ARG GRPC_VERSION=1.60.0
+# Default value should be aligned with upstream Selenium (https://github.com/SeleniumHQ/selenium/blob/trunk/java/maven_deps.bzl)
```
Comment (VietND96) [inline Base/Dockerfile:7]: Update dependencies version same as upstream

File: Base/Dockerfile [hunk 2]
```diff
@@ -115,7 +116,8 @@ COPY supervisord.conf /etc
 #==========
 RUN touch ${SEL_DIR}/config.toml \
   && wget --no-verbose https://github.com/SeleniumHQ/selenium/releases/download/${RELEASE}/selenium-server-${VERSION}.jar \
-    -O ${SEL_DIR}/selenium-server.jar
+    -O ${SEL_DIR}/selenium-server.jar \
+  && echo "${SEL_PASSWD}" > ${SEL_DIR}/initialPasswd
```
Comment (VietND96) [inline Base/Dockerfile:120]: In base image, we store the password under /opt/selenium/initialPasswd
Other images would refer the default password, only read from this file. There is no default value via ENV anymore

**Comment by VietND96** [comment]
> What is the actual change I should review? I see many unrelated things to the VNC password.

I have highlighted those via individual comments.

**Comment by diemol** [comment]
> > What is the actual change I should review? I see many unrelated things to the VNC password.
> 
> I have highlighted those via individual comments.

Thank you! 

In general, we try to have a commit/PR to contain a single set of related changes. Would be nice to follow that idea in the future.

**Review by diemol — APPROVED** [review]

**Comment by VietND96** [comment]
> > > What is the actual change I should review? I see many unrelated things to the VNC password.
> > 
> > 
> > I have highlighted those via individual comments.
> 
> Thank you!
> 
> In general, we try to have a commit/PR to contain a single set of related changes. Would be nice to follow that idea in the future.

Sure, thank you! Except only 1 this PR that mixed something else work-in-progress

**Pull request merged**